packages/oidc-auth/src/index.ts

```diff
@@ -60,13 +60,39 @@ type OidcAuthEnv = {
   OIDC_COOKIE_DOMAIN?: string
 }
 
+/**
+ * Configure the OIDC variables programmatically.
+ * If used, should be called before any other OIDC functions for the Hono context.
+ * Unconfigured values will fallback to environment variables.
+ */
+export const setOidcAuthEnv = (c: Context, config: Partial<OidcAuthEnv>) => {
+  const oidcAuthEnv = c.get('oidcAuthEnv')
+  if (oidcAuthEnv !== undefined) {
+    throw new HTTPException(500, { message: 'OIDC Auth env is already configured' })
+  }
+  const _ = getOidcAuthEnv(c, config)
```

Comment:
The only other thing I'd consider is moving the parts of `getOidcAuthEnv` that actually set the config into `setOidcAuthEnv`

Then have `getOidcAuthEnv` call `setOidcAuthEnv` when `oidcAuthEnv === undefined`

Reply:
Makes for a noisier diff but is a clearer separation of concerns, applied! 👍